airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -731,24 +733,42 @@ def __init__(self, replication_key: str, stream_slice_step: str = "P30D", **kwar
         super().__init__(**kwargs)
         self.replication_key = replication_key
         self._stream_slice_step = stream_slice_step
+        self._cursor = None
+
+    @property
+    def cursor(self) -> Optional[Cursor]:
```

Comment:
I hate this but stream creation is very far away from Cursor creation. I'll look at how I can get those two closer

Reply:
I expect it to be fairly difficult because of the way streams are instantiated...